### PR TITLE
Fix the issue that tiup may choose yanked version if it's already installed

### PR DESCRIPTION
**Pull request by lucklove**
### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

Release notes:

```release-note
NONE
```

### CODE REVIEW

File: pkg/environment/env.go [hunk 1]
```diff
@@ -36,6 +36,11 @@ const (
 	tiupName = "tiup"
 )
 
+var (
+	// ErrInstallFirst indicates that a component/version is not installed
+	ErrInstallFirst = errors.New("component not install")
```
Comment (AstroProfundis) [inline pkg/environment/env.go:41]:
```suggestion
	ErrInstallFirst = errors.New("component not installed")
```

File: pkg/environment/env.go [hunk 2]
```diff
@@ -179,17 +184,64 @@ func (env *Environment) downloadComponent(component string, version pkgver.Versi
...
-// DownloadComponentIfMissing downloads the specific version of a component if it is missing
-func (env *Environment) DownloadComponentIfMissing(component string, ver pkgver.Version) (pkgver.Version, error) {
-	versions, err := env.profile.InstalledVersions(component)
+func (env *Environment) SelectInstalledVersion(component string, ver pkgver.Version) (pkgver.Version, error) {
+	installed, err := env.Profile().InstalledVersions(component)
 	if err != nil {
 		return "", err
 	}
 
+	errInstallFirst := errors.Annotatef(ErrInstallFirst, "use `tiup install %s` to install component `%s` first", component, component)
```
Comment (AstroProfundis) [inline pkg/environment/env.go:193]: why not use `ErrInstallFirst` here?

File: pkg/environment/env.go [hunk 3]
```diff
@@ -200,23 +252,13 @@ func (env *Environment) DownloadComponentIfMissing(component string, ver pkgver.
...
-		for _, v := range versions {
-			if pkgver.Version(v) == ver {
-				installed = true
-				break
-			}
+	needDownload := false
+	if ver.IsEmpty() {
+		ver, err = env.SelectInstalledVersion(component, ver)
+		needDownload = errors.Cause(err) == ErrInstallFirst
+		if err != nil && errors.Cause(err) != ErrInstallFirst {
```
Comment (AstroProfundis) [inline pkg/environment/env.go:259]:
```suggestion
		if err != nil && !needDownload {
```

File: pkg/environment/env.go (hunk 2 above)
Comment (lucklove) [reply pkg/environment/env.go:193]: Because this error will be displayed to the user

**Pull request merged**